core/execution/special_orders.go

```diff
@@ -268,57 +248,183 @@ func (m *Market) updateLPOrders(
 	// from the list if the liquidity engine instructed to
-	// cancel them
-	var cancelEvts []events.Event
+	// cancel them, but also the list of all new orders to be created
 	for _, order := range allOrders {
-		if _, ok := distressedParties[order.Party]; ok {
-			// party is distressed, not processing
-			continue
-		}
+		order.UpdatedAt = now
 
 		// these order were actually cancelled, just send the event
 		if _, ok := cancelIDs[order.ID]; ok {
-			// cancelEvts = append(cancelEvts, events.NewOrderEvent(ctx, order))
-			// these orders were submitted exactly the same before,
-			// so there's no reason we would not be able to submit
-			// let's panic if an issue happen
-		} else {
-			// set the status to active again
-			order.Status = types.OrderStatusActive
-			conf, _, err := m.submitValidatedOrder(ctx, order)
-			if conf == nil || err != nil {
-				distressedOrders = append(distressedOrders, order)
-				distressedParties[order.Party] = struct{}{}
-			} else if len(conf.Trades) > 0 {
-				m.log.Panic("submitting liquidity orders after a reprice should never trade",
-					logging.Order(*order))
+			order.Status = types.OrderStatusCancelled
+			orderEvts = append(orderEvts, events.NewOrderEvent(ctx, order))
+			continue
+		}
+
+		subFn(order)
+	}
+
+	for _, order := range submits {
+		order.UpdatedAt = now
+		subFn(order)
+	}
+
+	// send cancel events
+	m.broker.SendBatch(orderEvts)
+
+	// an ordered list of positions
+	var (
+		positions     = make([]events.MarketPosition, 0, len(partiesPos))
+		marginsBefore = map[string]*num.Uint{}
+		id            = m.GetID()
+		assetID, _    = m.mkt.GetAsset()
+	)
+	// now we can check parties positions
+	for party, pos := range partiesPos {
+		positions = append(positions, pos)
+		mar, _ := m.collateral.GetPartyMarginAccount(id, party, assetID)
+		marginsBefore[party] = mar.Balance
+	}
+
+	sort.Slice(positions, func(i, j int) bool {
+		return positions[i].Party() < positions[j].Party()
+	})
+
+	// now we calculate all the new margins
+	risks := m.updateMargin(ctx, positions)
+	if len(risks) > 0 {
+		transfers, closed, bondPenalties, err := m.collateral.MarginUpdate(
+			ctx, m.GetID(), risks)
+		if err == nil && len(transfers) > 0 {
+			evt := events.NewLedgerMovements(ctx, transfers)
+			m.broker.Send(evt)
+		}
+
+		cancelled := m.applyBondPenaltiesAndLiquidationExcludingPending(
+			ctx, bondPenalties, closed, marginsBefore,
+		)
+
+		// now ensure we have all parties pending status updated
+		for _, v := range positions {
+			if m.liquidity.IsLiquidityProvider(v.Party()) {
+				if _, ok := cancelled[v.Party()]; !ok {
+					// this party LP wasn't cancelled, so it should be now
+					// not pending anymore,
+					m.liquidity.RemovePending(v.Party())
+				}
 			}
 		}
 
-		// if an auction has been started, we just break now
-		if m.as.InAuction() {
-			// enteredAuction = true
-			// auctionFrom = i
-			// break
-			return nil
+		_ = m.equityShares.SharesExcept(m.liquidity.GetInactiveParties())
```

Comment:
does this err?

Reply:
Nope, it returns a value that we don't need here.